reconcile/ocm_machine_pools.py

```diff
@@ -1,137 +1,400 @@
+
+
+def fetch_current_state(
+    ocm_map: OCMMap,
+    clusters: Iterable[ClusterV1],
+) -> Mapping[str, list[AbstractPool]]:
+    current_state: MutableMapping[str, list[AbstractPool]] = {}
     for cluster in clusters:
-        cluster_name = cluster["name"]
-        ocm = ocm_map.get(cluster_name)
-        machine_pools = ocm.get_machine_pools(cluster_name)
-        for machine_pool in machine_pools:
-            machine_pool["cluster"] = cluster_name
-            current_state.append(machine_pool)
+        current_state[cluster.name] = []
+        ocm = ocm_map.get(cluster.name)
+        if cluster.spec and cluster.spec.hypershift:
+            machine_pools = ocm.get_node_pools(cluster.name)
+            for machine_pool in machine_pools:
+                current_state[cluster.name].append(
+                    NodePool(
+                        id=machine_pool["id"],
+                        replicas=machine_pool["replicas"],
+                        aws_node_pool=AWSNodePool(
+                            instance_type=machine_pool["aws_node_pool"]["instance_type"]
+                        ),
+                        taints=machine_pool.get("taints"),
+                        labels=machine_pool.get("labels"),
+                        subnet=machine_pool.get("subnet"),
+                        cluster=cluster.name,
+                    )
+                )
+        else:
+            machine_pools = ocm.get_machine_pools(cluster.name)
+            for machine_pool in machine_pools:
+                current_state[cluster.name].append(
+                    MachinePool(
+                        id=machine_pool["id"],
+                        replicas=machine_pool["replicas"],
+                        instance_type=machine_pool["instance_type"],
+                        taints=machine_pool.get("taints"),
+                        labels=machine_pool.get("labels"),
+                        cluster=cluster.name,
+                    )
+                )
 
-    return ocm_map, current_state
+    return current_state
 
 
-def fetch_desired_state(clusters):
-    desired_state = []
+def create_desired_state_from_gql(
+    clusters: Iterable[ClusterV1],
+) -> DesiredStateList:
+    desired_state: DesiredStateList = DesiredStateList(cluster_pools=[])
     for cluster in clusters:
-        cluster_name = cluster["name"]
-        machine_pools = cluster["machinePools"]
-        for machine_pool in machine_pools:
-            machine_pool["cluster"] = cluster_name
-            labels = machine_pool.pop("labels")
-            if labels:
-                machine_pool["labels"] = json.loads(labels)
-            taints = machine_pool.pop("taints")
-            if taints:
-                machine_pool["taints"] = taints
-            desired_state.append(machine_pool)
+        if cluster.machine_pools:
+            is_hypershift = False
+            if cluster.spec:
+                is_hypershift = True if cluster.spec.hypershift else False
```

Comment:
Can be simplified by

```python
is_hypershift = cluster.spec is not None and bool(cluster.spec.hypershift)
```

Not sure if `cluster.spec` can be `None`, if it's not nullable in schema, just `bool(cluster.spec.hypershift` will be enough.

Reply:
Hm, I actually prefer the more verbose version :slightly_smiling_face: